Description: Backup and restore RUST_BACKTRACE env var in backtrace-optional test
 The test optional_backtrace_is_not_present_without_environment_variable()
 in tests/backtrace-optional.rs assumes that the RUST_BACKTRACE variable is
 unset, thus failing when calling RUST_BACKTRACE=1 cargo test. This change
 backups and unsets the environment variable before the check, and restores
 it afterwards.
Forwarded: https://github.com/shepmaster/snafu/pull/299
Author: Wolfgang Silbermayr <wolfgang@silbermayr.at>
Last-Update: 2021-08-22

--- a/tests/backtrace-optional.rs
+++ b/tests/backtrace-optional.rs
@@ -14,6 +14,16 @@
 
 #[test]
 fn optional_backtrace_is_not_present_without_environment_variable() {
+    // The check requires RUST_BACKTRACE to be unset. We back up the current
+    // value of the environment variable, so we can restore it afterwards.
+    let bt_env_name = "RUST_BACKTRACE";
+    let bt_env = std::env::var_os(bt_env_name);
+    std::env::remove_var(bt_env_name);
+
     let sometimes = BacktraceSometimes.build();
     assert!(ErrorCompat::backtrace(&sometimes).is_none());
+
+    if let Some(v) = bt_env {
+        std::env::set_var(bt_env_name, v);
+    }
 }
